### PR TITLE
Version Updates

### PR DESCRIPTION
**Pull request by NiklasRentzCAU**
Updates Sprotty to version 1.1.0 and alongside it now requires node 16, es2017, new webpack 5 and others.

### CODE REVIEW

File: applications/klighd-cli/webpack.config.js [hunk 1]
```diff
@@ -20,10 +20,16 @@ module.exports = {
...
+            "buffer": require.resolve("buffer"),
+            "crypto": require.resolve("crypto-browserify"),
+            "os": require.resolve("os-browserify/browser"),
+            "stream": require.resolve("stream-browserify"),
+        },
     },
 
     node: {
-        net: "mock",
+        __dirname: 'mock',
```
Comment (Eddykasp) [inline applications/klighd-cli/webpack.config.js:32]: inconsistent use of ' and " in this file

File: applications/klighd-vscode/README.md [hunk 2]
```diff
@@ -109,20 +109,19 @@ To intercept an action, your extension has to provide an action handler with the
...
 ```typescript
-// - refId: your registration id returned from the setLanguageClient command
 // - kind: the action kind that should be intercepted by the handler
 // - handler: the action handler that is called for the provided action type.
-vscode.commands.executeCommand("klighd-vscode.addActionHandler", refId: string, kind: string, handler: ActionHandler);
+vscode.commands.executeCommand("klighd-vscode.addActionHandler", kind: string, handler: ActionHandler);
 ```
 
 ### Dispatching Actions
+*This feature is currently not supported*
```
Comment (Eddykasp) [inline applications/klighd-vscode/README.md:124]: does this break any of our stuff?

File: applications/klighd-vscode/src/commandContributions.ts [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        vscode.window.onDidChangeActiveTextEditor(async editor => {
+            const activeWebview = manager.findActiveWebview()
+            const alreadyOpen = editor?.document.uri.scheme + "://" + editor?.document.uri.path === activeWebview?.diagramIdentifier?.uri
+            const shouldOpen = manager.storageService.getItem('diagramOpen')
+            if (editor && manager.getSyncWithEdior() && !alreadyOpen && shouldOpen) {
+                manager.openDiagram(editor.document.uri);
+            }
+        })
+    );
+}
```
Comment (Eddykasp) [inline applications/klighd-vscode/src/commandContributions.ts:125]: missing newline at end of file

File: applications/klighd-vscode/src/extension.ts [hunk 4]
```diff
@@ -108,51 +109,44 @@ export function activate(context: vscode.ExtensionContext): void {
...
             if (!isAction(action)) {
                 vscode.window.showErrorMessage(
                     `${command.addActionHandler} command called with invalid arguments. Please refer to the documentation for reference about the correct usage.`
                 );
                 return;
             }
-
-            extension.webviews.forEach((webview) => webview.dispatch(action));
+            // TODO: this has not been tested yet if the messenging change works like this.
+            messenger.sendNotification({ method: "diagram/accept" }, { type: 'webview', webviewType: diagramType }, action);
```
Comment (Eddykasp) [inline applications/klighd-vscode/src/extension.ts:138]: what exactly is affected by this and what needs to be done to test whether this works?

File: applications/klighd-vscode/src/webview-endpoint.ts [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+ *   + Department of Computer Science
+ *     + Real-Time and Embedded Systems Group
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+import "reflect-metadata";
```
Comment (Eddykasp) [inline applications/klighd-vscode/src/webview-endpoint.ts:17]: inconsistent use of ' and " in this file

File: applications/klighd-vscode/src/webview-endpoint.ts [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+            },
+            { sender: this.messageParticipant }
+        );
+    }
+
+    override async receiveAction(message: ActionMessage): Promise<void> {
+        await super.receiveAction(message);
+        await this.languageClient.sendNotification(acceptMessageType, message);
+    }
+}
```
Comment (Eddykasp) [inline applications/klighd-vscode/src/webview-endpoint.ts:92]: missing newline

File: applications/klighd-vscode/src/webview-panel-manager.ts [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+        }
+        // endpoint.
+        return endpoint as unknown as LspWebviewEndpoint;
+    }
+
+    protected override didCloseWebview(endpoint: WebviewEndpoint): void {
+        super.didCloseWebview(endpoint);
+        this.storageService.setItem('diagramOpen', false)
+    }
+}
```
Comment (Eddykasp) [inline applications/klighd-vscode/src/webview-panel-manager.ts:104]: missing newline

File: packages/klighd-core/src/diagram-pieces/actions.ts [hunk 8]
```diff
@@ -89,11 +89,11 @@ export class SetDiagramPieceCommand extends ResetCommand {
...
-    undo(_context: CommandExecutionContext): SModelRoot {
+    undo(_context: CommandExecutionContext): SModelRootImpl {
         return this.root
     }
 
-    redo(_context: CommandExecutionContext): SModelRoot {
+    redo(_context: CommandExecutionContext): SModelRootImpl {
         return this.root
     }
 }
```
Comment (Eddykasp) [inline packages/klighd-core/src/diagram-pieces/actions.ts:99]: missing newline

**Review by Eddykasp — COMMENTED** [review]
Quote mark formatting could be done in a separate dedicated PR, it's still a little unclear what part of the messaging is now broken and/or untested.

Is there anything else besides the bookmarks that is now disabled or no longer working?

File: applications/klighd-vscode/README.md (hunk 2 above)
Comment (NiklasRentzCAU) [reply applications/klighd-vscode/README.md:124]: This feature has not been used in any other extension, so it does not break anything either.

File: applications/klighd-vscode/src/extension.ts (hunk 4 above)
Comment (NiklasRentzCAU) [reply applications/klighd-vscode/src/extension.ts:138]: `// Command provided for other extensions to dispatch an action if a webview is open`
As this comment above suggests, this is the feature that I marked as currently not supported in the documentation. To test this we would need to use this feature in another extension, i.e. dispatch some action that should be executed from that other extension. As I don't see a use case currently, I think this is fine to document it being (potentially) not fully supported atm.

**Pull request merged**